WooCommerce/Classes/POS/Utils/POSProductProvider.swift

```diff
@@ -0,0 +1,90 @@
+import Foundation
+import Yosemite
+import protocol Storage.StorageManagerType
+import class WooFoundation.CurrencySettings
+
+/// Product provider for the Point of Sale feature
+///
+final class POSProductProvider {
+    private let storageManager: StorageManagerType
+    private var siteID: Int64
+    private var currencySettings: CurrencySettings
+
+    init() {
+        self.storageManager = ServiceLocator.storageManager
+        self.siteID = ServiceLocator.stores.sessionManager.defaultSite?.siteID ?? 0
+        self.currencySettings = ServiceLocator.currencySettings
+    }
```

Comment:
I opted for not passing any parameters through the initializer to keep the separation between `app` and `pos `, otherwise we would be exposing the app-layer types to where the `POSProductProvider` is instantiated. Also it would seem that at the moment we would only open these for testing purposes, since we do not declare any of these dependencies in the `pos` side.